### ISSUE TITLE
PE - No two lecture classes can have overlapping slots.

### ISSUE THREAD

**Issue by nowknowing** [nus-cs2030/2021-s1#152]
# Description 
At level 3, it is mentioned "No two lecture classes can have overlapping slots". 

However, in level 4 test cases, this now seems to means no two lecture classes of the SAME MODULE, can have overlapping slots. In other words, if DIFF module, two lectures can have overlapping slots.

Is this correct?

So in level 3,
Two lectures, diff instructor, diff venue, diff module code, same timing. ==> clashWith is true?

## Topic: 
Possible inconsistencies.
Vague

**Comment by lkldev** [comment]
Hi, which test case in level 3 are you referring to? from what I see all the lectures have the same venue.

**Comment by tlylt** [comment]
```java
Lecture hchia_L = new Lecture("CS2030", 1, "LT19", new Instructor("hchia"), 10)
jshell> hchia_L.clashWith(new Lecture("CS2040", 1, "LT19", new Instructor("tanst"), 10))
$.. ==> true
```
If this is the one that is confusing you, this test case in level3 shows that lectures happening at same time in the same **location** will clash, it doesn't say anything about whether lectures of different modules can overlap.

**Comment by nowknowing** [comment]
Yes, question is in level 3, 
what should clashWith output, given diff instructor, diff venue, diff mod code, but same timing?

**Comment by tlylt** [comment]
> Yes, question is in level 3,
> what should clashWith output, given diff instructor, diff venue, diff mod code, but same timing?

```java
CS2040 L1 @ I3-AUD [tanst] 15--17
CS2030 L2 @ LT19 [hchia] 16--18
```
This test case in Level4 says a lot about what you want to know.
P.S. Hi IkIdev you are fast:)

**Comment by lkldev** [comment]
I believe if there are different instructor, different venue, different module code and same timing logically it is not considered a clash

**Comment by nowknowing** [comment]
Yes I believe so now.
I have implemented such that [only timing same] outputs true i.e. clashes (during exam).  2030 shouldn't clash with 2040.
But upon reading level 4 (now) , I am realizing this should have given false. i.e. no clash.

I shall recreate a new check altogether in level 4, to account for diff mod code.

**Comment by nowknowing** [comment]
> I believe if there are different instructor, different venue, different module code and same timing logically it is not considered a clash

Unless its CS2030 and CS2040.

**Comment by Maurice2n97** [comment]
Unless the lecture codes are different i.e from different modules, then any clash in timing but with different venue and different lecturer should be no problem.
if their from the same module, then the lecture cannot clash no matter whehter the venues are different or not( likewise same for lecturer)

**Comment by charoi** [comment]
Yes agree with @Maurice2n97 , first check is the module code for mine

**Comment by marcuspeh** [comment]
> Unless the lecture codes are different i.e from different modules, then any clash in timing but with different venue and different lecturer should be no problem.
> if their from the same module, then the lecture cannot clash no matter whehter the venues are different or not( likewise same for lecturer)

Yup i believe the same as @Maurice2n97 too although I have got to admit level 3 phrasing kinda means otherwise
